### PR TITLE
Fine grained deps

### PR DESCRIPTION
**Pull request by gregmagolan**

### CODE REVIEW

**Review by alexeagle — APPROVED** [review]

File: integration/bazel/src/hello-world/BUILD.bazel [hunk 1]
```diff
@@ -18,39 +18,42 @@ ng_module(
...
     bootstrap = [
-        "//:node_modules/zone.js/dist/zone-testing-bundle.js",
-        "//:node_modules/reflect-metadata/Reflect.js",
+        "@npm//node_modules/zone.js:dist/zone-testing-bundle.js",
+        "@npm//node_modules/reflect-metadata:Reflect.js",
     ],
     browsers = [
         "@io_bazel_rules_webtesting//browsers:chromium-local",
-        "@io_bazel_rules_webtesting//browsers:firefox-local",
+        # TODO(gregmagolan): re-enable firefox testing once fixed
```
[the remark this answers is not in the document]
Comment (gregmagolan) [reply integration/bazel/src/hello-world/BUILD.bazel]: 👍

File: packages/compiler-cli/src/ngtsc/metadata/test/BUILD.bazel [hunk 2]
```diff
@@ -4,7 +4,7 @@ load("//tools:defaults.bzl", "jasmine_node_test", "ts_library")
 
 ts_library(
     name = "test_lib",
-    testonly = 1,
+    testonly = True,
```
[the remark this answers is not in the document]
Comment (gregmagolan) [reply packages/compiler-cli/src/ngtsc/metadata/test/BUILD.bazel]: done

File: integration/bazel/package.json [hunk 3]
```diff
@@ -4,20 +4,23 @@
   "version": "0.0.0",
   "license": "MIT",
   "dependencies": {
-    "reflect-metadata": "file:../../node_modules/reflect-metadata",
-    "rxjs": "file:../../node_modules/rxjs",
-    "tslib": "file:../../node_modules/tslib",
-    "zone.js": "file:../../node_modules/zone.js"
+    "reflect-metadata": "0.1.12",
```
[the remark this answers is not in the document]
Comment (alexeagle) [reply integration/bazel/package.json]: oh, hmm, we need a TODO somewhere to figure out how to keep these up-to-date then.

File: packages/BUILD.bazel [hunk 4]
```diff
@@ -11,6 +11,10 @@ load("//tools:defaults.bzl", "ts_library")
 ts_library(
     name = "types",
     srcs = glob(["*.ts"]),
+    deps = [
+        "@angular_deps//@types/hammerjs",
```
[the remark this answers is not in the document]
Comment (gregmagolan) [reply packages/BUILD.bazel]: done

File: integration/bazel/postinstall-patches.js [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+sed('-i', '(\'response\' in xhr)', '(\'response\' in (xhr as any))',
+    'node_modules/rxjs/src/observable/dom/AjaxObservable.ts');
+*/
+
+// TODO(gmagolan): remove these rxjs patches once
+// https://github.com/bazelbuild/rules_typescript/pull/293
+// lands and makes it into a rules_typescript release or when
+// https://github.com/ReactiveX/rxjs/pull/4213
+// lands in rxjs that applies these changes
+log('\n# patch: reactive/rxjs conform with tsetse check');
```
[the remark this answers is not in the document]
Comment (gregmagolan) [reply integration/bazel/postinstall-patches.js]: postinstall patches removed

File: integration/bazel/package.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (gregmagolan) [reply integration/bazel/package.json]: done

**Pull request merged**